tests/robustness/failpoints.go

```diff
@@ -103,6 +103,18 @@ func validateFailpoint(clus *e2e.EtcdProcessCluster, failpoint Failpoint) error
 	return nil
 }
 
+func addMinimalRevForCompactFailpoint(failpoint Failpoint, batchNum int) Failpoint {
+	switch failpoint.Name() {
+	case "compactAfterCommitBatch", "compactBeforeCommitBatch":
```

Comment:
Instead of using this decorator and doing name matching, can you add a flag?
```
var (
    CompactBeforeCommitBatchPanic            Failpoint = goPanicFailpoint{"compactBeforeCommitBatch", triggerCompact{waitForRevision: true}, AnyMember}
	CompactAfterCommitBatchPanic             Failpoint = goPanicFailpoint{"compactAfterCommitBatch", triggerCompact{waitForRevision: true}, AnyMember}
)

type triggerCompact struct {
 waitForRevision bool
}

	
func (t triggerCompact) Trigger(_ *testing.T, ctx context.Context, member e2e.EtcdProcess, _ *e2e.EtcdProcessCluster) error {
	cc, err := clientv3.New(clientv3.Config{
		Endpoints:            member.EndpointsGRPC(),
		Logger:               zap.NewNop(),
		DialKeepAliveTime:    10 * time.Second,
		DialKeepAliveTimeout: 100 * time.Millisecond,
	})
	if err != nil {
		return fmt.Errorf("failed creating client: %w", err)
	}
	defer cc.Close()
    if t.waitForRevision {
      	var rev int64
	  for {
		resp, err := cc.Get(ctx, "/")
		if err != nil {
			return err
		}
		if rev = resp.Header.Revision > member.Cfg.CompactionBatchLimit {
		    break
		}
		time.Sleep(50 * time.Millisecond)
	  }
    }

	_, err = cc.Compact(ctx, rev)
	if err != nil && !strings.Contains(err.Error(), "error reading from server: EOF") {
		return err
	}
	return nil
}
```